### PR TITLE
BUGFIX Page Type listed in Page Type dropdown when $can_be_root is false

### LINKED ISSUE #2430: BUG Page Type listed in Page Type dropdown when $can_be_root is false
If a page is on the root level and page type has `private static $can_be_root = false`, it shouldn't be included as a dropdown option.

### PR DESCRIPTION
**Pull request by muskie9**
resolves #2430

### CODE REVIEW

File: code/Model/SiteTree.php [hunk 1]
```diff
@@ -2567,6 +2567,11 @@ protected function getClassDropdown()
         foreach ($classes as $class) {
             $instance = singleton($class);
 
+            //if the current page is root and the instance can't be, exclude
+            if ($instance->config()->get('can_be_root') == false && $this->ParentID == 0) {
```
Comment (ScopeyNZ) [inline code/Model/SiteTree.php]:
```suggestion
            if (!$instance->config()->get('can_be_root') && $this->ParentID === 0) {
```

Comment (muskie9) [reply code/Model/SiteTree.php]: I'll try that. I couldn't seem to get tests running locally for whatever reason so I've been having to leverage travis, which isn't ideal for tracking down the issue.

Comment (ScopeyNZ) [reply code/Model/SiteTree.php]: This suggestion is the same functionally, just making strict comparisons where the should be 🙂

Comment (muskie9) [reply code/Model/SiteTree.php]: Still no luck with local test, so debugging the failure is proving to be difficult. At a glance does it look like my methodology of testing that the class is in the source array as a key?

Comment (ScopeyNZ) [reply code/Model/SiteTree.php]: Seems right - I don't have time to dive into it though. Sometimes with travis you'll have to reproduce the install steps to get the same failure. This'll mean `git clone silverstripe/silverstripe-cms`, checking out your PR branch, and then running the steps in `.travis.yml`.

Comment (muskie9) [reply code/Model/SiteTree.php]: No worries, I'll give that a try and see what I can come up with.

Comment (muskie9) [reply code/Model/SiteTree.php]: @ScopeyNZ that's gotten me a lot closer on local, thnx for the pointer 👏

**Comment by muskie9** [comment]
The failure was ultimately permissions for the user rather than the new config check. I should've noticed the other `$this->loginWithPermission('ADMIN');` calls.

**Comment by michalkleiner** [comment]
Merge this in? :)

**Pull request merged**